### DISCUSSION TITLE
graphql/scalars/docs/scalars/big-int

### DISCUSSION THREAD

**Discussion by giscus** [Urigo/graphql-scalars#1851] [Docs Discussions]
# graphql/scalars/docs/scalars/big-int

GraphQL Scalars Documentation

https://the-guild.dev/graphql/scalars/docs/scalars/big-int

**Comment by Hitlii** [comment]
Hello, good evening, excuse me, is there any example of how to implement this?

**Reply by GuilleEneas** [reply]
I would check how they test it in their own repo.

**Reply by youpiwaza** [reply]
Yeah it's kinda obscure.
You'll need to import from 'graphql-scalars', by prefixing the scalar type with 'GraphQL'.

Here's an example from one of my entities in NestJs (Typescript):

```ts
import { ObjectType, Field, ID } from '@nestjs/graphql';
//               v Don't forget to prefix type with "GraphQL"
import { GraphQLBigInt, GraphQLDate } from 'graphql-scalars';

@ObjectType()
export class GuyEntity {
  @Field(() => String, {
    description: 'The guy name.',
    nullable: false,
  })
  name: string;

  //                 v GraphQL scalar type
  @Field(() => GraphQLBigInt, {
    description: 'Social security number, must be unique.',
    nullable: false,
  })
  //                      v js/ts type
  socialSecurityNumber: number;

  @Field(() => GraphQLDate, {
    description: 'Date of creation',
    nullable: false,
  })
  createdAt: Date;
}
```

Also you need the package installed from npm/yarn/pnpm/bun/whatever ^^'

Cheers

**Reply by ChristianDavis** [reply]
I know that's an example, but SSN is NOT a number. It's a string, as it can start with 0.
